File: Java Food Order App/PremiumCustomer.java
package FoodOrderApp;

public class PremiumCustomer extends Customer implements Bill {
	float discount = 12.5f;
	int deliveryCharges = -100;
	float membershipCardDiscount = 7.5f;
	static String type = "Premium Customer";
	
	PremiumCustomer(byte customerID, Name customerName, long customerPhone, Address customerAddress, String customerEmail) {
		super(customerID, customerName, customerPhone, customerAddress, customerEmail);
	}
	
	public void payBill(float total) {
		System.out.println("\tTotal Price is: " + total);
		System.out.println("\n\tSince you are our " + PremiumCustomer.type);
		float billAmount = (total - (total * discount / 100)) + deliveryCharges +
				(total - (total * membershipCardDiscount / 100));
		System.out.println("\tAfter Discount: " + billAmount);
	}
}

File: Kotlin Food Order App/PremiumCustomer.kt
package foodOrderApp

class PremiumCustomer(
    customerID : Byte,
    customerName: Name,
    customerPhone: Long,
    customerAddress: Address,
    customerEmail: String
) : Customer(customerID.toByte(), customerName, customerPhone, customerAddress, customerEmail), BillOrder {

    private var discount: Float = 12.5f
    private var deliveryCharges: Int = -100
    private var membershipCardDiscount: Float = 7.5f
    companion object {
        var type = "Premium Customer"
    }
    override fun payBill(total: Float) {
        println("\tTotal Price is: $total")
        println("\n\tSince you are our $type")
        val billAmount = (total - (total * discount / 100)) + deliveryCharges -
                (total - total * membershipCardDiscount / 100)
        println("\tAfter Discount: $billAmount")
    }
}